README.md
# Dectorio
[![GitHub issues](https://img.shields.io/github/issues/jpanther/Dectorio.svg)](https://github.com/jpanther/Dectorio/issues) [![CrowdIn](https://d322cqt584bo4o.cloudfront.net/dectorio/localized.svg)](https://crowdin.com/project/dectorio) [![GitHub license](https://img.shields.io/github/license/jpanther/Dectorio.svg)](https://github.com/jpanther/Dectorio/blob/develop/LICENSE)

A simple mod that adds decorative elements to beautify your factory. New technology research is required to unlock the various items before they can be crafted and some default recipes are adjusted to better reflect how these items should be made.

Inspiration for the mod came from a lot of other mods on the Factorio mod portal, although all the assets in Dectorio have been custom made for this mod and are either modified versions of base game assets or created from scratch. If you'd like to use any of these in your own mod, please observe the GNU GPLv3 license.

Requires Factorio version 0.15 or later.

## How to install

Download the latest stable release from the [Factorio mod portal](https://mods.factorio.com/mods/PantherX/Dectorio) or at <https://github.com/jpanther/Dectorio/releases/latest> and copy the entire zip file to your Factorio mods directory:

* Windows: `%appdata%\Factorio\mods`
* Mac: `~/Library/Application Support/factorio/mods/`
* Linux: `~/.factorio/mods`

> **Note:** If playing with Factorio 0.15, please download v0.7.6 as this is the last stable release supported on 0.15. v0.8.0 and later are only supported on Factorio 0.16.

When you launch the game, the new mod will be enabled by default.

## Current Features

### Decorative items

* Adds traffic bollards for decorating roadways or protecting your factory from vehicular collisions

  ![Traffic bollard](https://img.pantherx.net/dectorio/traffic-bollard.png)

### Flooring styles

* Adds basic gravel paths (crafted from raw stone, coal, iron and copper)

  ![Gravel styles](https://img.pantherx.net/dectorio/gravel.png)

* Adds wooden floorboard tiles with sound effects
* Adds grid-based concrete that reflects the classic Factorio 0.15 style concrete

  ![Concrete and Wooden floorboards](https://img.pantherx.net/dectorio/concrete-wood.png)

* Adds painted concrete tiles - Hazard (black/yellow), Danger (red/yellow), Emergency (white/red), Caution (orange/white), Radiation hazard (yellow/pink), Defect (blue/white), Operations (black/white) and Safety (white/green)

  ![Painted concrete](https://img.pantherx.net/dectorio/painted-concrete.png)

* Painted Hazard concrete replaces default Hazard concrete^
* Increased flooring item stack sizes^

### Landscaping

* Adds items for placing dirt, sand, dark sand, grass, dry grass, red desert and dark red desert tiles
* Ability to place water and green water tiles (aka. 'waterfill')
* Ability to place decorative rocks

  ![Landscaping items](https://img.pantherx.net/dectorio/landscaping.png)

### Walls and Gates

* Adds items for wooden barricade, stone wall, chain-link fence and concrete walls
* Walls have varied resistances and strengths to allow for better progression
* Increased wall item stack sizes^

  ![Walls](https://img.pantherx.net/dectorio/walls.png)

* Adds items for hazard gates

  ![Gates](https://img.pantherx.net/dectorio/hazard-gates.png)

### Signs

* Adds placable wooden and steel signs with configurable icons

  ![Signs](https://img.pantherx.net/dectorio/signs.png)

### Signals & Lamps

* Adds a new Glowing lamp that will glow bright colours over a large radius when passed a coloured signal from the circuit network

  ![Glowing lamp](https://img.pantherx.net/dectorio/glowing-lamp.png)

* Adds new signal colours - purple, orange, tangerine and aqua
* Adds signal colours for raw resources, smelted items, fluids, science packs, inserters, belts, splitters and circuits

  ![New signal colours](https://img.pantherx.net/dectorio/signal-colours.png)

* Adds new icon based signals for making it easier to understand your circuit network logic
* New signal icons for base game signal colours that better represent the actual signal colour

  ![Signal selection](https://img.pantherx.net/dectorio/signals.png)

### General

* Includes settings for enabling or disabling each of the above features (Options > Mods > Startup). Ideally these should only be changed before starting a new game. Changes on existing games will require content to be migrated and some items may be lost if you disable components you've already used on an existing save. Always make a backup first!

  ![Mod settings](https://img.pantherx.net/dectorio/settings.png)

* Items above marked ^ can be further customised in the mod settings dialog

Check out the [Mod Spotlight video by Xterminator](https://www.youtube.com/watch?v=4OOs8o0dnug) to see the mod in action (note that this features an older version of Dectorio without some of the features mentioned above).

## Development

### Contributing

Feel free to get in touch with any issues or suggestions for new features you'd like to see. I also welcome pull requests to fix bugs or implement new features. The best way to make contact is via [GitHub Issues](https://github.com/jpanther/Dectorio/issues), however you can alternatively post on the [Factorio Mod Portal](https://mods.factorio.com/mods/PantherX/Dectorio/discussion) (but please note that it may take longer to get a response this way as there are no notifications of new messages on the portal and I only check it sporadically). Usually when I find issues reported on the Mod Portal, I'll open a new GitHub issue to track its progress.

#### **Did you find a bug?**

* **Ensure the bug was not already reported** by searching on GitHub under [Issues](https://github.com/jpanther/Dectorio/issues).
  > **Note:** If you find a **Closed** issue that seems like it is the same thing that you're experiencing, open a new issue and include a link to the original issue in the body of your new one.

* If you're unable to find an open issue addressing the problem, [open a new one](https://github.com/jpanther/Dectorio/issues/new). Be sure to include a **title and clear description**, as much relevant information as possible, and a **code sample** or a **test case** demonstrating the expected behaviour that is not occurring.

#### **Can you write a patch that fixes a bug or do you want to implement a new feature?**

* Fork the `develop` branch.
  > **Note:** The `master` branch is reserved for releases and is always in a 'production-ready' state. The `develop` branch is where the latest changes are developed and tested ahead of the next release.

* Create a new branch in your fork that describes the issue or new feature (i.e. `bugfix-nastybug` or `feature-mynewfeature`)

* Make the changes required to fix the bug or implement the new feature.

* Pull the upstream `develop` branch and merge it into your fork regularly to stay up to date with changes.

* When you're ready to submit your changes, and there are no conflicts with the upstream branch, push your fork and open a new GitHub pull request to the `develop` branch with your patch.

* Ensure the PR description clearly describes what the patch does. If it fixes a reported bug or feature request, include the relevant issue number.

### Language Support

Dectorio uses CrowdIn to translate the mod into various languages.

Fully supported languages:
* English

Community translations:
* Chinese Simplified (30%)
* Chinese Traditional (36%)
* Czech (38%)
* German (100%)
* Korean (100%)
* Polish (71%)

If you are able to translate the mod into any of the above languages, or you know an additional language that you'd like to add a new translation for, please visit the [Dectorio project on CrowdIn](https://crowdin.com/project/dectorio) to help with the translation effort.

### Mod Compatibility

**Note:** If playing with Factorio 0.15 saves, please download Dectorio v0.7.6 as this is the last release supported on 0.15. Dectorio v0.8.0 and later are only supported on Factorio 0.16.

Limited testing has been completed with other mods. Warnings will be shown in-game to highlight any known issues. If you come across a specific compatibility issue, please let me know and I'll see what I can do to fix it.

It is not recommended to use other coloured signal mods at the same time as Dectorio. This can cause coloured signals to be overwritten multiple times and lead to game-breaking compatibility issues with other mods. If you would like more colours, you can add these in the Dectorio config.lua file (instructions are in the file).

Dectorio includes specific support for [Alien Biomes](https://mods.factorio.com/mods/Earendel/alien-biomes).

Other mods with additional Dectorio support (thanks!): [Concreep](https://mods.factorio.com/mods/Mylon/Concreep), [FARL](https://mods.factorio.com/mods/Choumiko/FARL), [Pavement Drive Assist](https://mods.factorio.com/mods/Arcitos/PavementDriveAssist).

Other decorative mods worth checking out (all compatible with Dectorio): [Asphalt Roads](https://mods.factorio.com/mods/Arcitos/AsphaltRoads), [Beautiful Bridge Railway](https://mods.factorio.com/mods/kapaer/beautiful_bridge_railway), [Color Coding](https://mods.factorio.com/mods/justarandomgeek/color-coding), [Naked Rails](https://mods.factorio.com/mods/futileohm/naked-rails).

Mods with known compatibility issues with Dectorio (either don't install these, or edit your mod settings and disable the conflicting components in Dectorio): [Expanded Color Lamps](https://mods.factorio.com/mods/Klonan/Expanded_Color_Lamps), [More Colours](https://mods.factorio.com/mods/zx64/more-colours), [More signal color](https://mods.factorio.com/mods/devilwarriors/More%20signal%20color), [Concrete and Stone](https://mods.factorio.com/mods/Earendel/concrete-and-stone), [Reinforced Walls](https://mods.factorio.com/mods/ChurchOrganist/Reinforced-Walls), [No Walls or Gates](https://mods.factorio.com/mods/abregado/no-walls), [Picker Tweaks](https://mods.factorio.com/mods/Nexela/PickerTweaks).

### Changelog

Refer to the [CHANGELOG](changelog.txt) for a detailed list of changes in each version. The changelog is also available in-game from the _Mods_ screen.

### Credits

Developed with ❤ by [James Panther](https://github.com/jpanther) in Melbourne, Australia.  
Special thanks to [marcfj](https://github.com/marcfj) for support and play testing.

#### Translators

German translation by [seeba8](https://github.com/seeba8) and [ST-DDT](https://crowdin.com/profile/ST-DDT).  
Polish translation by [Tomirad](https://github.com/Tomirad).  
Czech translation by [dereksp](https://crowdin.com/profile/dereksp).  
Korean translation by [Xagros](https://crowdin.com/profile/GimoXagros).  


CHANGELOG.md
# Changelog

The latest stable release is available at <https://github.com/jpanther/Dectorio/releases/latest>

## v0.8.0 - 2017-12-15

### Added
* Factorio 0.16 support
* New Painted concrete textures to match new vanilla Concrete style
* Painted concrete floorings now support high resolution graphics
* New landscaping items for placing all the new base game terrain types
* Graphics support for placing stone and concrete walls alongside water (others to come in a future update)
* Support for new in-game changelog feature
* More robust technology recipe-unlock handling to improve compatibility with third-party mods
* Support for migrating Fire hazard concrete from _Color Coding_ mod to Emergency concrete

### Removed
* Factorio 0.15 support - use Dectorio v0.7.6 when playing 0.15 saves
* Removed support for _Water Fix_ mod as the base game now supports placing tiles alongside water
* Redundant assets to reduce overall mod size

### Locales
* Updated German translations

## v0.7.6 - 2017-12-01

### Added
* New setting for toggling whether ores can be placed as gravel
* Dectorio floorings can now be placed adjacent to water when _Water Fix_ mod is installed

### Bugfixes
* Fixed that water tiles were not available for crafting when playing with _Alien Biomes_

### Locales
* Updated German, Korean and Polish translations

## v0.7.5 - 2017-11-21

### Added
* New crafting tab to separate decorative items from other crafting menus - can be disabled in settings (thanks to [Aidiakapi](https://github.com/Aidiakapi))
* New localisation system - help translate Dectorio by visiting [CrowdIn](https://crowdin.com/project/dectorio)
* Added initial Czech translation (thanks to [dereksp](https://crowdin.com/profile/dereksp))

### Bugfixes
* Fixed some audio and repairability issues with traffic bollards

## v0.7.4 - 2017-11-04

### Bugfixes
* Fixed technology error that appeared when Decoratives setting was set to disabled

## v0.7.3 - 2017-10-27

### Added
* New gfx variants for chain-link fence so it looks more realistic

### Bugfixes
* Major fixes to make signs more robust in multiplayer games
    * Signs are now indestructible and unminable while the player that placed the sign is selecting an icon
    * Icon selection GUI now has a 'Cancel' button
    * Only one sign can be placed at a time (by the same player), preventing blank signs from being placed
    * Icons are now indestructible and can only be removed by removing the sign
    * Icons are now correctly removed when a sign is destroyed, preventing orphaned icons from being left on the game surface

## v0.7.2 - 2017-10-26

### Added
* New gravel styles - Iron, Copper and Coal Gravel (placed using raw iron, copper and coal ores)
* New _Traffic bollard_ decorative item (unlocked via _Plastics_ technology)
* Landscaping technology now allows placement of decorative rock entities
* Better handling for when third party mods that affect Dectorio are added or removed

### Bugfixes
* Added some additional locale strings to fix issues with _Creative Mode_ mod

## v0.7.1 - 2017-10-21

### Bugfixes
* Fixed migration error for existing saves without _Alien Biomes_ mod

## v0.7.0 - 2017-10-20

Please take note of the concrete recipe changes in this release. You may need to adjust parts of your factory.

### Added
* New in-game settings for configuring Dectorio (replaces the old settings that were configured via config.lua)
* Landscaping support for _Alien Biomes_ mod

### Changed
* Concrete recipe no longer requires iron sticks and reverts to the vanilla iron ore recipe
* Concrete walls now require iron sticks to craft
* Adjusted flooring layer priority so they appear correctly when placed adjacent to each other (order is now ground > gravel > stone path > wood floor > concrete)
* Tweaked minimap colours for some flooring styles

## v0.6.2 - 2017-10-10

### Added
* Polish translation (thanks to [Tomirad](https://github.com/Tomirad))

## v0.6.1 - 2017-10-05

### Changed
* Reverted Landfill crafting time back to vanilla recipe

### Bugfixes
* Fixed error in migration script that would cause a crash when painted concrete was disabled
* Fixed that signs would render icons incorrectly when placed on alternate surfaces (ie. when using _Factorissimo_ mod)
* Fixed GUI bug that would crash headless server when placing signs with certain other mods enabled

## v0.6.0 - 2017-07-21

### Added
* Wooden and steel signs (unlocked via new _Signs_ technology)
* _Glowing lamp_ which will glow bright colours over a large radius when passed a signal from the circuit network
* Added more variations to the edge gfx for the gravel tiles
* New icon based signals for easier circuit network logic
* Signal colour for Steam

### Changed
* Hazard concrete now places as the vanilla hazard-concrete tile (fixes blueprint imports and issues when Dectorio is removed)
* Hazard concrete keeps the Dectorio look, however you can set a config.lua flag to change it to use the vanilla look, if preferred
* Balancing - halved crafting time of painted concrete

## v0.5.14 - 2017-07-18

### Added
* German translation (thanks to [seeba8](https://github.com/seeba8))

## v0.5.13 - 2017-07-12

### Bugfixes
* Notifications will no longer warn about incompatible mods when those components of Dectorio are disabled

## v0.5.12 - 2017-06-01

### Added
* New painted concrete types - Caution (orange/white), Danger (red/yellow), Defect (blue/white) and Operations (black/white)

### Changed
* Adjusted radiation and safety concrete variants to be less bright

### Bugfixes
* Fixed hazard-right-side variant not correctly masked
* Fixed painted concrete on different layer to normal concrete

## v0.5.11 - 2017-05-31

### Added
* New flooring - Gravel (placed using raw stone)
* Stone wall can now be upgraded to concrete wall

### Changed
* Adjusted map colours for flooring types to better distinguish the floors
* Landscaping items will no longer appear in blueprints (configurable in config.lua)

config.lua
-- Dectorio configuration

local DECT = {}
DECT.CONFIG = {}

-- CONFIGURABLE OPTIONS
-- ---------------------------------
-- These options can be changed to suit individual play-style
-- If playing multiplayer, ensure the same config.lua is used on each player's system
-- ---------------------------------

DECT.CONFIG.SIGNALS = {
	-- All the signal colors to be changed/added
	-- RGBA values from 0 to 1, alpha is optional
	-- The order of the virtual signals is the order they will appear in-game
	-- NOTE: New virtual signals will need localised names to be provided in '/locale/[language]/dectorio.cfg'
	-- Color names are from Name That Color (http://chir.ag/projects/name-that-color/)
	{type="virtual", name="signal-red", color={r=1.00,g=0.16,b=0.10,a=1.00}},           -- Scarlet
	{type="virtual", name="signal-orange", color={r=0.99,g=0.44,b=0.22,a=1.00}},        -- Orange
	{type="virtual", name="signal-tangerine", color={r=1.00,g=0.58,b=0.14,a=1.00}},     -- Tree Poppy
	{type="virtual", name="signal-yellow", color={r=1.00,g=0.96,b=0.27,a=1.00}},        -- Lemon Yellow
	{type="virtual", name="signal-green", color={r=0.00,g=0.95,b=0.17,a=1.00}},         -- Free Speech Green
	{type="virtual", name="signal-cyan", color={r=0.01,g=0.98,b=1.00,a=1.00}},          -- Aqua
	{type="virtual", name="signal-aqua", color={r=0.05,g=0.67,b=0.99,a=1.00}},          -- Spiro Disco Ball
	{type="virtual", name="signal-blue", color={r=0.07,g=0.35,b=0.98,a=1.00}},          -- Crayon Blue
	{type="virtual", name="signal-purple", color={r=0.65,g=0.38,b=0.99,a=1.00}},        -- Lavender Indigo
	{type="virtual", name="signal-pink", color={r=1.00,g=0.42,b=0.99,a=1.00}},          -- Ultra Pink
	{type="virtual", name="signal-white", color={r=1.00,g=1.00,b=1.00,a=1.00}},         -- White
	{type="virtual", name="signal-grey", color={r=0.80,g=0.80,b=0.80,a=1.00}},          -- Iron
	{type="virtual", name="signal-black", color={r=0.22,g=0.13,b=0.56,a=1.00}},         -- Blue Gem

	{type="item", name="raw-wood", color={r=0.57,g=0.36,b=0.12,a=0.86}},                -- Afghan Tan
	{type="item", name="coal", color={r=0.17,g=0.17,b=0.17,a=0.86}},                    -- Bunker
	{type="item", name="stone", color={r=0.64,g=0.50,b=0.30,a=0.86}},                   -- Muesli
	{type="item", name="iron-ore", color={r=0.24,g=0.80,b=0.99,a=0.86}},                -- Picton Blue
	{type="item", name="iron-plate", color={r=0.63,g=0.64,b=0.64,a=0.86}},              -- Grey Chateau
	{type="item", name="steel-plate", color={r=0.80,g=0.80,b=0.80,a=0.86}},             -- Iron
	{type="item", name="copper-ore", color={r=0.91,g=0.58,b=0.13,a=0.86}},              -- Carrot Orange
	{type="item", name="copper-plate", color={r=0.92,g=0.59,b=0.46,a=0.86}},            -- Dark Salmon
	{type="item", name="uranium-ore", color={r=0.69,g=0.96,b=0.18,a=0.86}},             -- Green Yellow
	{type="item", name="science-pack-1", color={r=0.71,g=0.05,b=0.09,a=1.00}},          -- Cornell Red
	{type="item", name="science-pack-2", color={r=0.18,g=0.60,b=0.11,a=1.00}},          -- Forest Green
	{type="item", name="science-pack-3", color={r=0.15,g=0.65,b=0.75,a=1.00}},          -- Blue Green
	{type="item", name="military-science-pack", color={r=0.23,g=0.27,b=0.31,a=1.00}},   -- Limed Spruce
	{type="item", name="production-science-pack", color={r=0.58,g=0.11,b=0.76,a=1.00}}, -- Dark Orchid
	{type="item", name="high-tech-science-pack", color={r=0.81,g=0.69,b=0.08,a=1.00}},  -- Galliano
	{type="item", name="space-science-pack", color={r=0.96,g=0.96,b=0.96,a=1.00}},      -- White Smoke
	{type="item", name="transport-belt", color={r=0.90,g=0.75,b=0.10,a=0.92}},          -- Sunflower
	{type="item", name="underground-belt", color={r=0.90,g=0.75,b=0.10,a=0.92}},        -- Sunflower
	{type="item", name="splitter", color={r=0.90,g=0.75,b=0.10,a=0.92}},                -- Sunflower
	{type="item", name="fast-transport-belt", color={r=0.84,g=0.08,b=0.02,a=0.92}},     -- Rosso Corsa
	{type="item", name="fast-underground-belt", color={r=0.84,g=0.08,b=0.02,a=0.92}},   -- Rosso Corsa
	{type="item", name="fast-splitter", color={r=0.84,g=0.08,b=0.02,a=0.92}},           -- Rosso Corsa
	{type="item", name="express-transport-belt", color={r=0.02,g=0.60,b=0.83,a=0.92}},  -- Rich Electric Blue
	{type="item", name="express-underground-belt", color={r=0.02,g=0.60,b=0.83,a=0.92}},-- Rich Electric Blue
	{type="item", name="express-splitter", color={r=0.02,g=0.60,b=0.83,a=0.92}},        -- Rich Electric Blue
	{type="item", name="electronic-circuit", color={r=0.18,g=0.60,b=0.00,a=0.92}},      -- Napier Green
	{type="item", name="advanced-circuit", color={r=0.72,g=0.00,b=0.00,a=0.92}},        -- Guardsman Red
	{type="item", name="processing-unit", color={r=0.27,g=0.26,b=1.00,a=0.92}},         -- Neon Blue
	{type="item", name="burner-inserter", color={r=0.38,g=0.31,b=0.26,a=0.92}},         -- Saddle
	{type="item", name="inserter", color={r=0.78,g=0.51,b=0.20,a=0.92}},                -- Golden Bell
	{type="item", name="long-handed-inserter", color={r=0.76,g=0.20,b=0.13,a=0.92}},    -- Dark Pastel Red
	{type="item", name="fast-inserter", color={r=0.27,g=0.45,b=0.58,a=0.92}},           -- Jelly Bean
	{type="item", name="filter-inserter", color={r=0.45,g=0.27,b=0.54,a=0.92}},         -- Affair
	{type="item", name="stack-inserter", color={r=0.56,g=0.63,b=0.16,a=0.92}},          -- Citron
	{type="item", name="stack-filter-inserter", color={r=0.70,g=0.70,b=0.69,a=0.92}},   -- Magnesium

	{type="fluid", name="water", color={r=0.00,g=0.43,b=0.75,a=0.68}},                  -- French Blue
	{type="fluid", name="crude-oil", color={r=0.10,g=0.10,b=0.10,a=0.68}},              -- Rangoon Green
	{type="fluid", name="heavy-oil", color={r=0.63,g=0.05,b=0.00,a=0.68}},              -- Dark Candy Apple Red
	{type="fluid", name="light-oil", color={r=0.71,g=0.42,b=0.00,a=0.68}},              -- Ginger
	{type="fluid", name="petroleum-gas", color={r=0.38,g=0.13,b=0.38,a=0.68}},          -- Palatinate Purple
	{type="fluid", name="sulfuric-acid", color={r=0.94,g=0.82,b=0.13,a=0.68}},          -- Broom
	{type="fluid", name="lubricant", color={r=0.00,g=0.75,b=0.30,a=0.68}},              -- Dark Pastel Green
	{type="fluid", name="steam", color={r=0.92,g=0.92,b=0.92,a=0.68}},                  -- Gallery
}

-- ---------------------------------
-- STOP HERE! DO NOT CHANGE BELOW THIS LINE!
-- ---------------------------------


-- INTERNAL CONFIGURATION
-- ---------------------------------

-- In-game configurable settings
DECT.ENABLED = {
	["signals"] = settings.startup["dectorio-signals"].value,
	["walls"] = settings.startup["dectorio-walls"].value,
	["landscaping"] = settings.startup["dectorio-landscaping"].value,
	["wood-floor"] = settings.startup["dectorio-flooring"].value,
	["concrete"] = settings.startup["dectorio-flooring"].value,
	["gravel"] = settings.startup["dectorio-gravel"].value,
	["painted-concrete"] = settings.startup["dectorio-painted-concrete"].value,
	["signs"] = settings.startup["dectorio-signs"].value,
	["decoratives"] = settings.startup["dectorio-decoratives"].value,
	["item-group"] = settings.startup["dectorio-item-group"].value,
}

-- The subgroups for all items in the mod
DECT.ITEM_GROUP = DECT.ENABLED["item-group"] and "dectorio" or "logistics"

DECT.CONFIG.SETTINGS = {
	["decorative_removal_probability"] = settings.startup["dectorio-decorative-removal-probability"].value,
	["vanilla_hazard_concrete"] = settings.startup["dectorio-vanilla-hazard-concrete-style"].value,
	["landscaping_blueprintable"] = settings.startup["dectorio-landscaping-blueprintable"].value,
	["flooring_stack_size"] = settings.startup["dectorio-flooring-stack-size"].value,
	["landscaping_stack_size"] = settings.startup["dectorio-landscaping-stack-size"].value,
	["walls_stack_size"] = settings.startup["dectorio-walls-stack-size"].value,
}

DECT.CONFIG.PAINT_VARIANTS = {
	{name="danger", color={r=0.81,g=0.31,b=0.31}},
	{name="emergency", color={r=0.86,g=0.36,b=0.38}},
	{name="caution", color={r=0.85,g=0.56,b=0.26}},
	{name="radiation", color={r=0.86,g=0.56,b=0.78}},
	{name="defect", color={r=0.47,g=0.50,b=1.00}},
	{name="operations", color={r=0.37,g=0.37,b=0.37}},
	{name="safety", color={r=0.61,g=0.77,b=0.40}},
}
DECT.CONFIG.GRAVEL_VARIANTS = {
	{name="stone", color={r=0.57,g=0.57,b=0.57}},
	{name="iron-ore", color={r=0.45,g=0.54,b=0.64}},
	{name="copper-ore", color={r=0.64,g=0.46,b=0.45}},
	{name="coal", color={r=0.33,g=0.33,b=0.33}},
}
DECT.CONFIG.SIGN_CATEGORIES = {"item", "fluid", "tool", "ammo", "armor", "capsule", "gun", "mining-tool", "module"}
DECT.CONFIG.SIGN_BLACKLIST = {"barrel", "loader", "simple%-entity", "player%-port", "computer", "coin", "small%-plane", "railgun", "vehicle%-machine%-gun", "tank%-machine%-gun", "factory%-1%-", "factory%-2%-", "factory%-3%-"}
DECT.CONFIG.BASE_TILES = {"dry-dirt", "dirt-1", "dirt-2", "dirt-3", "dirt-4", "dirt-5", "dirt-6", "dirt-7", "sand-1", "sand-2", "sand-3", "grass-1", "grass-2", "grass-3", "grass-4", "red-desert-0", "red-desert-1", "red-desert-2", "red-desert-3"}
DECT.CONFIG.BASE_WATER_TILES = {"water", "deepwater", "water-green", "deepwater-green"}
DECT.CONFIG.BASE_TREES = {"tree-01", "tree-02", "tree-03", "tree-04", "tree-05", "tree-06", "tree-07", "tree-08", "tree-09", "dead-dry-hairy-tree", "dead-grey-trunk", "dead-tree-desert", "dry-hairy-tree", "dry-tree"}
DECT.CONFIG.BASE_ROCKS = {"rock-huge", "rock-big", "sand-rock-big"}

DECT.INCOMPATIBLE = {}
DECT.INCOMPATIBLE.MODS = {
	["Expanded_Color_Lamps"] = {
		name = "Expanded_Color_Lamps",
		component = "signals",
		reason = "signals"
	},
	["more-colours"] = {
		name = "more-colours",
		component = "signals",
		reason = "signals"
	},
	["More signal color"] = {
		name = "More signal color",
		component = "signals",
		reason = "signals"
	},
	["Reinforced-Walls"] = {
		name = "Reinforced-Walls",
		component = "walls",
		reason = "tech"
	},
	["concrete-and-stone"] = {
		name = "concrete-and-stone",
		component = "walls",
		reason = "tech"
	},
	["no-walls"] = {
		name = "no-walls",
		component = "walls",
		reason = "disable"
	},
	["PickerTweaks"] = {
		name = "PickerTweaks",
		component = "signals",
		reason = "item"
	},
}

return DECT
